src/isomorphic/classic/element/ReactElement.js

```diff
@@ -83,8 +88,25 @@ var ReactElement = function(type, key, ref, self, source, owner, props) {
         writable: true,
         value: false,
       });
+      // self and source are DEV only properties.
+      Object.defineProperty(element, '_self', {
+        configurable: false,
+        enumerable: false,
+        writable: false,
+        value: self,
+      });
+      // Two elements created in two different places should be considered
+      // equal for testing purposes and therefore we hide it from enumeration.
+      Object.defineProperty(element, '_source', {
+        configurable: false,
+        enumerable: false,
+        writable: false,
+        value: source,
+      });
     } else {
-      this._store.validated = false;
+      element._store.validated = false;
+      element._self = self;
+      element._source = source;
```

Comment:
I used a single underscore here for consistency with `_owner` and `_store` while the properties in `config` use the double underscore.